airflow/assets/__init__.py

```diff
@@ -301,10 +328,14 @@ def __init__(
         self.uri = _sanitize_uri(_validate_non_empty_identifier(self, fields["uri"], uri))
         self.group = _validate_identifier(self, fields["group"], group) if group else self.asset_type
         self.extra = _set_extra_default(extra)
+        self.watchers = watchers or []
 
     def __fspath__(self) -> str:
         return self.uri
 
+    def __hash__(self) -> int:
```

Comment:
It was removed after discussing with @uranusjr . seems to introduce more trouble than benefit then. is there any reason we want to add it here?

Reply:
I am sure I needed it when implementing it but I just tested it without it and it worked so I'll remove it